### PR TITLE
Add console-server readiness/liveness probe

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Enhancement / new feature

### Description

Adds a readiness/liveness probe to the console-server to ensure that it is minimally functional.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: pkg/controller/consoleservice/consoleservice_controller.go [hunk 1]
```diff
@@ -738,7 +738,27 @@ func applyDeployment(consoleservice *v1beta1.ConsoleService, deployment *appsv1.
...
+				Command: []string{
+					"sh",
+					"-c",
+					"/probe.sh",
+				},
+			},
+		}
+
+		container.ReadinessProbe = &corev1.Probe{
+			InitialDelaySeconds: 30,
```
Comment (lulf) [inline pkg/controller/consoleservice/consoleservice_controller.go]: Could it be worth making the init delay configurable in the console service
CR? Maybe not important in the short term.

Comment (k-wall) [reply pkg/controller/consoleservice/consoleservice_controller.go]: We need a consistent way of exposing this sort of thing for configuration.   In the infraconfigs we've added a `podTemplate` field, but this creates a ugly overlap between it and our CRD's own fields (e.g `podTemplate.container.resources` and `admin.resources`) so I am not sure this is the right pattern.    I wonder about reusing smaller kubernetes types instead.  For instance in this case we could add:

consoleServer.readinessProbe and consoleServer.livenessProbe which internally would reuse the Probe type and let the user override the timeouts.   WDYT?

Comment (lulf) [reply pkg/controller/consoleservice/consoleservice_controller.go]: Agreed, I think that strikes a good balance between ease of use and reusing kubernetes patterns at the appropriate level of abstraction.

Comment (k-wall) [reply pkg/controller/consoleservice/consoleservice_controller.go]: Ok, I'll take this approach. I'll merge this as it stands and incorporate the change into my next PR (which is pushes a number of variables to the CRD for configuration).

**Pull request merged**